src/ui/patch_hub/patchset_details_and_actions.sh

```diff
@@ -145,16 +150,22 @@ function handle_bookmark_action()
   local -n _patchset="$1"
   local raw_patchset="$2"
   local output
+  local loading_pid
+  local ret
 
   output=$(download_series "${_patchset['patchset_url']}" "${lore_config['save_patches_to']}")
   if [[ "$?" != 0 ]]; then
     create_message_box 'Error' 'Could not download patchset:'$'\n'"- ${_patchset['patchset_title']}"$'\n'"[error message] ${output}"
   fi
 
-  create_loading_screen_notification 'Bookmarking patchset'$'\n'"- ${_patchset['patchset_title']}"
+  create_async_loading_screen_notification 'Bookmarking patchset'$'\n'"- ${_patchset['patchset_title']}" &
+  loading_pid="$!"
 
   add_patchset_to_bookmarked_database "${raw_patchset}" "${lore_config['save_patches_to']}"
-  if [[ "$?" != 0 ]]; then
+  ret="$?"
+  stop_async_loading_screen_notification "$loading_pid"
+
```

Comment:
Nitpick alert: delete this extra blank line to maintain the "pattern" you used in the other files.